R/translate-sql-string.R

```diff
@@ -0,0 +1,33 @@
+# R prefers to specify start / stop or start / end
+# databases usually specify start / length
+# https://www.postgresql.org/docs/current/functions-string.html
+#' @export
+#' @rdname sql_variant
+sql_substr <- function(f = "SUBSTR") {
+  function(x, start, stop) {
+    start <- as.integer(start)
+    length <- pmax(as.integer(stop) - start + 1L, 0L)
+
+    sql_call2(f, x, start, length)
+  }
+}
+
+# if no parameters are provided, str_sub returns the whole string
+# if just start is provided, str_sub returns the rest of the string (after start)
+#' @export
+#' @rdname sql_variant
+sql_str_sub <- function(f = "SUBSTR", full_length = "drop", compute_method = "LENGTH") {
+  function(string, start = 1L, end = -1L) {
+    start <- as.integer(start)
```

Comment:
I think I'll implement this a little differently, so can you please just make sure that the unit tests cover the positive and negative cases that you are thinking about? (The tests would live in `tests/testthat/test-translate-sql-string.R`)